tests/test_eternal_guard.py

```diff
@@ -0,0 +1,147 @@
+#!/usr/bin/env python
+#
+# License: BSD
+#   https://raw.githubusercontent.com/splintered-reality/py_trees/devel/LICENSE
+#
+##############################################################################
+# Imports
+##############################################################################
+
+import py_trees
+import py_trees.console as console
+
+##############################################################################
+# Eternal Guard
+##############################################################################
+
+
+def create_tasks():
+    return [
+        py_trees.behaviours.Count(
+            name="R-R-S",
+            fail_until=0,
+            running_until=2,
+            success_until=100,
+            reset=True
+        ),
+        py_trees.behaviours.Count(
+            name="R-S",
+            fail_until=0,
+            running_until=1,
+            success_until=100,
+            reset=True
+        ),
+    ]
+
+
+def impl_eternal_guard_checks(name, root, eternal_guard, tasks):
+    console.banner(name)
+
+    py_trees.tests.print_assert_banner()
+
+    py_trees.tests.tick_tree(root, 1, 1, print_snapshot=True)
+    print(console.green + "Tick 1: first guard fails, eternal guard fails" + console.reset)
+    py_trees.tests.print_assert_details("eternal_guard", py_trees.common.Status.FAILURE, eternal_guard.status)
+    assert(eternal_guard.status == py_trees.common.Status.FAILURE)
+
+    py_trees.tests.tick_tree(root, 2, 2, print_snapshot=True)
+    print(console.green + "Tick 2: guard checks ok, task sequence is running" + console.reset)
+    py_trees.tests.print_assert_details("eternal_guard", py_trees.common.Status.RUNNING, eternal_guard.status)
+    assert(eternal_guard.status == py_trees.common.Status.RUNNING)
+
+    py_trees.tests.tick_tree(root, 3, 5, print_snapshot=True)
+    print(console.green + "Tick 5: guards still ok, task sequence finished" + console.reset)
+    py_trees.tests.print_assert_details("eternal_guard", py_trees.common.Status.SUCCESS, eternal_guard.status)
+    assert(eternal_guard.status == py_trees.common.Status.SUCCESS)
+
+    py_trees.tests.tick_tree(root, 6, 7, print_snapshot=True)
+    print(console.green + "Tick 7: tasks are running again, but the first guard fails" + console.reset)
+    py_trees.tests.print_assert_details("eternal_guard", py_trees.common.Status.FAILURE, eternal_guard.status)
+    assert(eternal_guard.status == py_trees.common.Status.FAILURE)
+    for task in tasks:
+        py_trees.tests.print_assert_details(task.name, py_trees.common.Status.INVALID, task.status)
+        assert(task.status == py_trees.common.Status.INVALID)
+
+
+def test_eternal_guard_idiom():
+    root = py_trees.composites.Selector(name="Root")
+    conditions = [
+        py_trees.behaviours.Count(
+            name="F-S-S-S-S-S-F",  # TODO: F-R-S-S-S-S-F once I have that working
```

Comment:
You have a TODO here that you might want to address in this PR / Remove?

Reply:
Thanks for the catch...that TODO is working already, just enabled it in the test.